### PR TITLE
ci: deploy website independently of library

### PR DESCRIPTION
**Pull request by juliodialpad**
## Description

Separated the deploy workflow in two:

- deploy.yml: Builds and deploys the docsite on every push to `production`, `alpha` or `beta`.
- release.yml: Builds and releases the library on every push to `production`, `alpha` or `beta` that contains a commit with changes in library files.

## Pull Request Checklist

 - [x] Ask the contributors if a similar effort is already in process or has been solved.
 - [x] Review the [contribution guidelines](https://github.com/dialpad/dialtone/blob/staging/.github/CONTRIBUTING.md).
 - [x] Use `staging` as your pull request's base branch. (All PRs using `production` as its base branch will be declined).
 - [x] Ensure all `gulp` scripts successfully compile.
 - [x] Update, remove, or extend all affected documentation.
 - [x] Ensure no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).
 - [x] Request a review from Brad Paugh, David Becher, or Drew Chandler.

### CODE REVIEW

File: .github/RELEASING.md [hunk 1]
```diff
@@ -13,21 +13,24 @@
...
 
 1. Make sure your `staging` and `production` branches are up-to-date locally. You should be in the `staging` branch. If you want to make a prerelease, it should be `beta` or `alpha` branch instead of `staging`.
 2. Stop your local server and keep your working directory clean before versioning.
 3. In your CLI window, run `npm run release` from the Dialtone repository directory.
-4. The script will update the `package.json` and `package-lock.json` files with the version number according to the types of changes introduced since the last release and will add release notes in the `CHANGELOG.MD` file.
-5. A release commit and a git tag associated with this commit will be created and pushed to the remote.
+4. If there are changes that should trigger a release:
+   - The script will update the `package.json` and `package-lock.json` files with the version number according to the types of changes introduced since the last release and will add release notes in the `CHANGELOG.MD` file. 
+   - A release commit and a git tag associated with this commit will be created and pushed to the remote.
+5. Otherwise, go to next section.
```
Comment (braddialpad) [inline .github/RELEASING.md:24]: Maybe mention somewhere in this section that you can still deploy changes to the docsite even if a library release was not triggered

Comment (braddialpad) [inline .github/RELEASING.md]: Maybe mention somewhere in this section that you can still deploy changes to the docsite even if a library release was not triggered

File: .github/RELEASING.md [hunk 2]
```diff
@@ -13,21 +13,24 @@
...
 
 ---
 
 **If you have made a production release:**
 
 6. We are ready to deploy the release. Switch to the `production` branch: `git checkout production`.
 7. Merge the release commits from `staging` using [`fast-forward` strategy](https://git-scm.com/docs/git-merge#Documentation/git-merge.txt---ff-only): `git merge staging --ff-only`.
-8. If the commits are merged correctly, it's ready to deploy and publish this release by pushing to the remote: `git push`. This will trigger the deploy through a Github Action.
+8. If the commits are merged correctly, it's ready to deploy the documentation site and/or publish this release by pushing to the remote: `git push`. This will trigger the deploy through a Github Action.
+9. If there are changes to library, this will trigger the release through a Github Action.
```
Comment (braddialpad) [inline .github/RELEASING.md]: Lets simplify this a bit

8. If the commits merged correctly, we can now push to the remote: `git push`.
9. If there were changes to the library, GitHub Actions will now deploy a new release of Dialtone to npm.
10. If there were changes to the documentation site, GitHub Actions will now deploy them to our GitHub Pages site: [dialpad.design](https://dialpad.design).
11. You should be able to see your deployment running at https://github.com/dialpad/dialtone/actions.

Comment (braddialpad) [inline .github/RELEASING.md:33]: Lets simplify this a bit

8. If the commits merged correctly, we can now push to the remote: `git push`.
9. If there were changes to the library, GitHub Actions will deploy a new release of Dialtone to npm.
10. If there were changes to the documentation site, GitHub Actions will deploy them to our GitHub Pages site.

**Comment by juliodialpad** [comment]
Did an extra clean-up after your suggestions, I think is more understandable now, what do you think? @braddialpad

**Review by braddialpad — APPROVED** [review]

**Pull request merged**